RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizer_cfi.py

```diff
@@ -3,6 +3,10 @@
 from RecoLocalTracker.SiPixelClusterizer.SiPixelClusterizerDefault_cfi import SiPixelClusterizerDefault as _SiPixelClusterizerDefault
 siPixelClusters = _SiPixelClusterizerDefault.clone()
 
+# *only for the cms-patatrack repository*
+# ensure reproducibility for CPU <--> GPU comparisons
+siPixelClusters.payloadType = "HLT"
```

Comment:
somewhat contrary to my naive expectations, the round of tests which succeeded revealed a large amount of regressions in Run1 and Run2 offline legacy workflows: see [here](https://github.com/cms-sw/cmssw/pull/31721#issuecomment-712479697), (already buried deep down the stack of gitHub notifications).
This change in the source of the gains payload (`"HLT"`=per column as opposed to `"Offline"`=per pixel) looks like a plausible source of the regression. Given the comment above, can you clarify if this change is intended or is a leftover of earlier passes?

Reply:
@mmusich thanks for catching this.
Indeed, as the comment says this change was not supposed to slip in the production release in a way that affect other workflows.

Out of my head I don't remember if that's *required* to run the Patatrack workflow, or if we set it like this for an easier comparison with the HLT performance.

Reply:
I would say it should not be a requirement to run the patatrack workflow, but only a way of getting cleaner comparison of the  GPU workflow (so using the new clusterization algorithm, presumably coded with the HLT case in mind) w.r.t. CPU.

Reply:
It is a requirement

Reply:
only the HLT payload has a CUDA interface

Reply:
and only the HLT payload has an efficient interface

Reply:
OK, so
  - we need `siPixelClusters.payloadType = "HLT"` for the patatrack workflows (on GPU, and on CPU to get consistent results)
  - we don't want to impact the other workflows

Suggestions how to make the switch ?

Reply:
But is `siPixelClusters` run on the GPU patatrack workflow at all?
Doing:
```bash
cmsrel CMSSW_11_2_0_pre7_Patatrack
cd CMSSW_11_2_0_pre7_Patatrack/src/
cmsenv
runTheMatrix.py -l 11634.502 -j 0
cd 11634.502_TTbar_14TeV+2021_Patatrack_PixelOnlyGPU+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST/
edmConfigDump step3_RAW2DIGI_RECO_VALIDATION_DQM.py > dump.py
```
I see:

```python
process.siPixelClustersPreSplitting = SwitchProducerCUDA(
cpu = cms.EDProducer("SiPixelClusterProducer",
ChannelThreshold = cms.int32(10),
ClusterMode = cms.string('PixelThresholdClusterizer'),
ClusterThreshold = cms.int32(4000),
ClusterThreshold_L1 = cms.int32(2000),
ElectronPerADCGain = cms.double(135),
MissCalibrate = cms.bool(True),
Phase2Calibration = cms.bool(False),
Phase2DigiBaseline = cms.double(1200),
Phase2KinkADC = cms.int32(8),
Phase2ReadoutMode = cms.int32(-1),
SeedThreshold = cms.int32(1000),
SplitClusters = cms.bool(False),
VCaltoElectronGain = cms.int32(1),
VCaltoElectronGain_L1 = cms.int32(1),
VCaltoElectronOffset = cms.int32(0),
VCaltoElectronOffset_L1 = cms.int32(0),
maxNumberOfClusters = cms.int32(-1),
mightGet = cms.optional.untracked.vstring,
payloadType = cms.string('HLT'),
src = cms.InputTag("siPixelDigis")
),
cuda = cms.EDAlias(
siPixelDigisClustersPreSplitting = cms.VPSet(cms.PSet(
type = cms.string('SiPixelClusteredmNewDetSetVector')
))
)
)
```

So it seems to me in the CUDA case it runs a different algo.
If that's the case, can the statement `siPixelClusters.payloadType = "HLT"` just be encapsulated in the SoA on CPU process-modifier (`customizePixelTracksSoAonCPU` IIRC) ?

Reply:
> can the statement `siPixelClusters.payloadType = "HLT"` just be encapsulated in the SoA on CPU process-modifier (`customizePixelTracksSoAonCPU` IIRC) ?

Most likely yes - I'll give it a try.

Reply:
Moved to `customizePixelTracksSoAonCPU` as suggested via https://github.com/cms-patatrack/cmssw/pull/560 .